src/lib/transaction/transactionGroupHelper.ts

```diff
@@ -0,0 +1,51 @@
+import type { IGroupsFromParam, ITransactionGroup } from './ITxParams';
+import type { OperationComponentTypes } from './operations/OperationComponentTypes';
+
+export default function groupComponents(
+    operations: typeof OperationComponentTypes[],
+    groups: IGroupsFromParam[],
+): (typeof OperationComponentTypes | ITransactionGroup)[] {
+    const lastGroup = groups[groups.length - 1];
+    const transactionGroups: (typeof OperationComponentTypes | ITransactionGroup)[] = [];
+
+    if (groups.length === 0) {
+        console.log("A group of operations wasn't provided");
+        return operations;
+    } else if (lastGroup && !operations[lastGroup.to]) {
```

Comment:
if groups.length cannot be 0, then lastGroup will always be truthy, therefore it makes no sense in this context.

```suggestion
    } else if (!operations[lastGroup.to]) {
```